File: libs/shared-types/python-src/debrief/types/tools/tool_list_response.py
"""ToolListResponse Pydantic model for listing available tools."""

from pydantic import BaseModel, Field
from typing import List, Optional
from .tool import Tool


class ToolListResponse(BaseModel):
    """Response format for listing available tools."""

    tools: List[Tool] = Field(
        ...,
        description="Array of available tools"
    )
    version: Optional[str] = Field(
        None,
        description="Optional version identifier for the tool collection"
    )
    description: Optional[str] = Field(
        None,
        description="Optional description of the tool collection"
    )

    class Config:
        extra = "forbid"  # No additional properties

File: libs/shared-types/src/zod/tools/tool_list_response.ts
import { z } from "zod"

export default z.object({ "tools": z.array(z.any()).describe("Array of available tools"), "version": z.union([z.string(), z.null()]).describe("Optional version identifier for the tool collection").default(null), "description": z.union([z.string(), z.null()]).describe("Optional description of the tool collection").default(null) }).strict().describe("Response format for listing available tools.")
